functions/Test-DbaLastBackup.ps1

```diff
@@ -393,7 +428,19 @@ Restores data and log files to alternative locations and only restores databases
 								}
 							}
 							
-							if ($destserver.Databases[$dbname] -ne $null -and !$NoDrop)
+							# Cleanup BackupFiles if -copyDestination and backup was moved to destination
+							if($copyDestination -eq $true -and $($lastbackup.path).startswith($($destserver.BackupDirectory)))
+							{
+								Write-Message -Level Verbose -Message "Removing backup file from $destination"
+								Remove-item $($lastbackup.fullname)
```

Comment:
File IO can fail. Try/catch the remove-item calls (`-ErrorAction Stop`, so it catches) and use Write-Message with the -ErrorReport parameter and warning level to report it